practices/java-jdbc/main/java/com/uabc/dowa/jdbc/repository/StudentRepository.java

```
@@ -1,23 +1,88 @@
+        List <Student> allStuds= new ArrayList<Student>();
+        pstmt = conn.prepareStatement("SELECT * FROM student");
+        try {
+            rs = pstmt.executeQuery();
+
+            while(rs.next()) {
+                student.setFirstName(rs.getString(2));
+                student.setLastName(rs.getString(3));
+                allStuds.add(student);
+            }
+        } catch (SQLException e) {
+            System.err.println("SQLException: " + e.getMessage());
+        }
+        finally {
+            conn.close();
+        }
+        return allStuds;
     }
 
-    public Student findStudentByFirstName(String firstName) {
-        return null;
+    public Student findStudentByFirstName(String firstName) throws SQLException{
+        Connection conn = ConnectionFactory.getConnection();
+        PreparedStatement pstmt = null;
+        ResultSet rs = null;
+        Student found = new Student();
+        try {
+            pstmt = conn.prepareStatement("SELECT * FROM student WHERE first_name = '" + firstName + "'");
+            rs = pstmt.executeQuery();
+            if(rs.next()) {
+                found.setFirstName(rs.getString(2));
+                found.setLastName(rs.getString(3));
+            }
+        } catch (SQLException e){
+            System.err.println("SQLException: " + e.getMessage());
+        }
+        finally {
+            conn.close();
+        }
+        return found;
     }
 
-    public void deleteStudent(Student student) {
+    public void deleteStudent(Student student) throws SQLException{
+        Connection conn = ConnectionFactory.getConnection();
+        PreparedStatement pstmt = null;
 
+        try {
+            pstmt = conn.prepareStatement("DELETE FROM student WHERE first_name = '"+student.getFirstName()+
+                    "' AND last_name = '"+student.getLastName()+"'");
+            pstmt.execute();
+        } catch (SQLException e) {
+            System.err.println("SQLException: " + e.getMessage());
```

Bad practice to eat the exception, it's oj to log it but you need to do something about it a simple RuntimeException would be ok